src/picosvg/svg.py

```diff
@@ -290,7 +290,7 @@ def _combine_clip_paths(self, clip_paths) -> SVGPath:
             raise ValueError("Cannot combine no clip_paths")
         if len(clip_paths) == 1:
             return clip_paths[0]
-        return intersection([c.as_cmd_seq() for c in clip_paths])
+        return intersection(clip_paths)
```

Comment:
this seems like a bug that should have failed a test but evidently didn't; we should add a test

Reply:
AH! Now I see what's going on here. I had misunderstood the `SVGShape.as_cmd_seq()` method, the function signature says it returns an `svg_meta.SVGCommandSeq`, which is an alias of `Iterable[SVGCommand]`. I forgot that `SVGPath` class has an `__iter__` method that yields exactly SVGCommands; i thought `as_cmd_seq` would return something like a list of tuples (one per command), but it returns an SVGPath! Which is also an SVGShape, and thus `svg_types.intersection` function works both ways (in the original, and in my modified copy in the line above).

I think as_cmd_seq() function should be changed so that its return value is typed as SVGPath, not the generic `svg_meta.SVGCommandSeq`. Return values should be strict and particular, input parameters can be wide.

Reply:
so basically we _are_ already testing this line, and it does the correct thing whether we pass clip_paths directly, or make that list comprehension with as_cmd_seq() (which also returns SVGPaths, not generic sequences).

Reply:
I believe return as SVGCommandSeq to be the correct return type because it makes clear what you can count on being there. Return as SVGPath introduces questions around how other attributes are handled that aren't relevant to an SVGCommandSeq.